ts/src/profiler.ts

```diff
@@ -72,6 +75,29 @@ export interface RequestProfile {
   labels?: {instance?: string};
 }
 
+/**
+ * @return number indicated by backoff if the response indicates a backoff and
+ * that backoff is greater than 0. Otherwise returns undefined.
+ */
+// tslint:disable-next-line: no-any
+function getServerResponseBackoff(response: any): number|undefined {
+  if (response && response.body && response.body.error &&
+      response.body.error.details &&
+      response.body.error.details instanceof Array) {
+    for (let i = 0; i < response.body.error.details.length; i++) {
+      const item = response.body.error.details[i];
+      if (typeof item === 'object' && item.retryDelay &&
+          typeof item.retryDelay === 'string') {
```

Comment:
Same comment about structurally malformed responses from the service. Should we log?

Reply:
Would this logging be used to indicate how the response is malformed?
If so, it's pretty easy to reproduce the error which we would expect to specify a backoff (just start about 20 copies of a trivial application with matching configurations and wait), which would make figuring out how something is malformed less important.

Reply:
> If so, it's pretty easy to reproduce the error which we would expect to specify a backoff

That sounds like e2e test. Excellent! Let's add one.